http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskServiceInterp.scala

```diff
@@ -192,6 +200,16 @@ class DiskServiceInterp[F[_]: Parallel](config: PersistentDiskConfig,
   ): F[Vector[ListPersistentDiskResponse]] =
     for {
       ctx <- as.ask
+
+      // throw 403 if user doesn't have project permission
+      hasProjectPermission <- cloudContext.traverse(cc =>
```

Comment:
so the traverse here basically means that the following only executes if `cloudContext` isn't `None`, and it evaluates to `None` otherwise. So `hasProjectPermission` can be None, true or false

The next bit `!hasProjectPermission.getOrElse(true)` ensures that if hasProjectPermission is:
- true --> no error is raised
- false -->  error is raised
- None --> no error is raised, then yeah the rest of the function decides which disks the user can see based on resource and project permissions

The goal of this bit is when a user is listing by project, they are immediately rejected if they don't have project permission (they shouldn't even see an empty list)

Reply:
Isn't there a use case in Terra for listing all visible disks independent of cloudContext? That's what the Environments page does, right?

Reply:
yeah so when `cloudContext` is `None`, then it doesn't execute the traverse (ie it doesn't do the auth check) and `hasProjectPermission` will therefore equal `None`, the exception won't be raised and the rest of the function executes like nothing happened